packages/smooth_app/lib/knowledge_panel/knowledge_panels/knowledge_panel_page.dart

```diff
@@ -59,4 +68,33 @@ class _KnowledgePanelPageState extends State<KnowledgePanelPage>
       return '';
     }
   }
+
+  Future<bool> _refreshProduct(BuildContext context) async {
+    try {
+      if (InheritedDataManager.of(context).currentBarcode.isNotEmpty) {
```

Comment:
I don't understand what `InheritedDataManager.of(context).currentBarcode` means. Isn't there an easier way to get the barcode?

Reply:
I use this InheritedWidget for sending the product to refresh through the widget tree without passing parameters, the widget tree is too big to do it by passing parameter 😅

Reply:
I understand the use-case, but we would be better off using a provider like we do in the rest of the app (cf. `main.dart`), wouldn't we? Unless there's a very good reason not to, but as there are no comments it's hard to figure out.